### PR TITLE
Add an option to use the external labels as selectors for the remote read endpoint

### PR DESCRIPTION
**Pull request by DrAuYueng**
Resolves https://github.com/prometheus/prometheus/issues/10158.

Signed-off-by: DrAuYueng <ouyang1204@gmail.com>

### CODE REVIEW

**Review by cstyan — COMMENTED** [review]
We should include a test here.

**Comment by DrAuYueng** [comment]
@cstyan Updated.

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -2868,6 +2868,10 @@ tls_config:
 
 # Configure whether HTTP requests follow HTTP 3xx redirects.
 [ follow_redirects: <boolean> | default = true ]
+
+# Whether ignore external_labels, will keep reading data that 
+# has already been written if true.
+[ ignore_external_labels: <boolean> | default = false ]
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:2874]:
```suggestion
[ filter_external_labels: <boolean> | default = true ]
```

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -2868,6 +2868,10 @@ tls_config:
 
 # Configure whether HTTP requests follow HTTP 3xx redirects.
 [ follow_redirects: <boolean> | default = true ]
+
+# Whether ignore external_labels, will keep reading data that
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:2872]: # Whether to use the external labels as selectors for the remote read endpoint.

**Review by roidelapluie — COMMENTED** [review]
I have added some wording suggestions. I also propose to make this a positive flag.

**Review by roidelapluie — APPROVED** [review]

**Review by cstyan — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**